src/Foreign/Lua/Module/System.hs

```diff
@@ -205,3 +228,47 @@ with_tmpdir parentDir tmpl callback =
       -- all args given. Second value must be converted to a string.
       tmpl' <- Lua.peek (fromAnyValue tmpl)
       Temp.withTempDirectory parentDir tmpl' (invokeWithFilePath callback')
+
+--
+-- Path manipulations
+--
+
+-- | See @System.FilePath.takeDirectory
+take_directory :: FilePath -> Lua FilePath
+take_directory fp = return $ Fp.takeDirectory fp
+
+-- | See @System.FilePath.takeFilename
+take_filename :: FilePath -> Lua FilePath
+take_filename fp = return $ Fp.takeFileName fp
+
+-- | See @System.FilePath.takeExtensions
+take_extensions :: FilePath -> Lua String
```

Comment:
`Fp.takeExtension` has type `FilePath -> FilePath`, so a type of `FilePath -> Lua FilePath` would seem more appropriate here (though those are equivalent).

Reply:
How did you see, that do you have code completion? I am in vs code. 
When I look at the docs its: String
https://hackage.haskell.org/package/filepath-1.4.2.1/docs/System-FilePath-Posix.html#v:takeExtensions

Reply:
You are right, my bad. I was looking at `takeExtensions`.